internal/controller/pkg/signature/attestation.go

```diff
@@ -0,0 +1,184 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Note(turkenh): This file is copied from https://github.com/sigstore/cosign/blob/ad478088320a3c04a96b3c183bbde2205fff7bbb/pkg/policy/attestation.go#L59
+// with little modification to remove the dependency on "github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
+// which brings in a lot of dependencies. Keeping the original license above.
```

Comment:
If possible it's nice to commit the unmodified file, then modify it in a separate commit. That way we can use git history to keep track of how the file has diverged from upstream.

Reply:
Done: https://github.com/crossplane/crossplane/pull/6022/commits/3dac6a167389615b5b02b0c2f6e8c57c00fa2279#diff-bb78683fbdec4d8ff2e1b2220f312298d3c02bb7bfbc032aa530afdd605b80de

If link gets broken with rebase etc, it is in the commit named "Simplify attestation verification skipping policy controller"